mono/utils/mono-compiler.h

```diff
@@ -113,5 +113,16 @@ typedef SSIZE_T ssize_t;
 #define MONO_GNUC_VERSION (__GNUC__ * 10000 + __GNUC_MINOR__ * 100 + __GNUC_PATCHLEVEL__)
 #endif
 
+/* Used to tell clang's ThreadSanitizer to not report data races that occur within a certain function */
+#if defined(__has_feature)
```

Comment:
@vargaz are you for or against @alexrp's idea? Generally non of the two versions look too magical to me - the current one is more compact (IMO), the idea of @alexrp has a better reusability which will come handy when working with other clang features in the future - generally I'm tempted to use what @alexrp suggested :)